### DISCUSSION TITLE
Better default ACLs

### DISCUSSION THREAD

**Discussion by jacoborus** [directus/directus#12830] [Feature Requests]
At the moment of this writing, Directus returns 403s on CRUD operations, both when the item we target is not found, or the user does not have permission to perform that action. Having in mind the security motivations as explained in this issue https://github.com/directus/directus/issues/9366, and also this proposal to allow to configure that behavior https://github.com/directus/directus/discussions/4368, I believe we can have better defaults for the ACLs

**Motivations**

The actual configuration it's useful when writing apps that need such a level of security, but for most of the publishing platforms (blogs, forums...) it's a bit of a deal-breaker because in those cases we need to know the difference between a 404 and a 403.

This behavior is confusing if the table uses UUIDs as PKs because if someone knows that hash, returning a 403 means the permissions have changed, not that the item was deleted as it should.

Public tables returning 403s, super admins getting 403s is quite contradictory.

**Change proposals**

Public tables:

- Never return 403s when reading

Private tables:

- Return 403s when the user has **no permission** to read the table
- Return 404s when the user has **full permission** to read the table, but the item does not exist

The 'Strictness control' described here: https://github.com/directus/directus/discussions/4368 is also important, but it would only apply if the user has permission to read **some** rows of the table, **but not all**.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
